code/analyticsedge/src/test/java/com/adobe/marketing/mobile/AnalyticsExtensionTests.java

```diff
@@ -421,7 +429,278 @@ public void test_handleAnalyticsTrackEvent_WhenPrivacyStatusIsNull() {
+        HashMap<String, Object> consequence = new HashMap<>();
+        consequence.put(AnalyticsConstants.EventDataKeys.ID, "id");
+        consequence.put(AnalyticsConstants.EventDataKeys.DETAIL, detailMap);
+        EventData eventData = new EventData();
+        eventData.putObject(AnalyticsConstants.EventDataKeys.TRIGGERED_CONSEQUENCE, consequence);
+        Event sampleEvent = new Event.Builder("rule event", EventType.RULES_ENGINE, EventSource.RESPONSE_CONTENT).setData(eventData).build();
+        setupPrivacyStatusInSharedState("optedin");
+        String timestamp = String.valueOf(sampleEvent.getTimestampInSeconds());
+
+        // test
+        analyticsExtension.handleRulesEngineEvent(sampleEvent);
+
+        // verify
+        ArgumentCaptor<Event> argument = ArgumentCaptor.forClass(Event.class);
+        PowerMockito.verifyStatic(MobileCore.class, times(0));
+        MobileCore.dispatchEvent(argument.capture(), (ExtensionErrorCallback<ExtensionError>) eq(null));
+    }
+
+    @Test
+    public void test_handleRulesEngineTrackEvent_IncorrectTypeInConsequence() {
+        //setup MobileCore mock method
+        PowerMockito.mockStatic(MobileCore.class);
+
+        // setup
+        HashMap<String, String> contextData = new HashMap<>();
+        contextData.put("key1", "value1");
+        contextData.put("key2", "value2");
+        HashMap<String, Object> detailMap = new HashMap<>();
+        detailMap.put("contextdata", contextData);
+        HashMap<String, Object> consequence = new HashMap<>();
+        consequence.put(AnalyticsConstants.EventDataKeys.ID, "id");
+        consequence.put(AnalyticsConstants.EventDataKeys.TYPE, "invalid");
+        consequence.put(AnalyticsConstants.EventDataKeys.DETAIL, detailMap);
+        EventData eventData = new EventData();
+        eventData.putObject(AnalyticsConstants.EventDataKeys.TRIGGERED_CONSEQUENCE, consequence);
+        Event sampleEvent = new Event.Builder("rule event", EventType.RULES_ENGINE, EventSource.RESPONSE_CONTENT).setData(eventData).build();
+        setupPrivacyStatusInSharedState("optedin");
+        String timestamp = String.valueOf(sampleEvent.getTimestampInSeconds());
+
+        // test
+        analyticsExtension.handleRulesEngineEvent(sampleEvent);
+
+        // verify
+        ArgumentCaptor<Event> argument = ArgumentCaptor.forClass(Event.class);
+        PowerMockito.verifyStatic(MobileCore.class, times(0));
+        MobileCore.dispatchEvent(argument.capture(), (ExtensionErrorCallback<ExtensionError>) eq(null));
+    }
+
+    @Test
+    public void test_handleRulesEngineTrackEvent_NoEventData() {
+        //setup MobileCore mock method
+        PowerMockito.mockStatic(MobileCore.class);
+
+        // setup
+        Event sampleEvent = new Event.Builder("rule event", EventType.RULES_ENGINE, EventSource.RESPONSE_CONTENT).build();
+        setupPrivacyStatusInSharedState("optedin");
+        String timestamp = String.valueOf(sampleEvent.getTimestampInSeconds());
```

Comment:
what is this timestamp for?

Reply:
whoops, leftover from editing the tests. removed them.